hexitor/src/editor.cpp

```diff
@@ -27,6 +27,33 @@
 #include "version.h"
 #include "hex_ctl.h" // For CP_UTF8
 #include <farkeys.h>
+#include <algorithm> // for std::max
+
+// Helper to get the length of a UTF-8 character from its first byte
+static int get_utf8_char_len(BYTE b)
+{
+	if ((b & 0x80) == 0) return 1;
+	if ((b & 0xE0) == 0xC0) return 2;
+	if ((b & 0xF0) == 0xE0) return 3;
+	if ((b & 0xF8) == 0xF0) return 4;
+	return 1; // Invalid start byte, treat as 1 byte
+}
+
+// Helper to find the start of the previous UTF-8 character
+UINT64 find_prev_utf8_start(const editor* ed, UINT64 current_offset)
+{
```

Comment:
suggest add a sane limit for backshifting, 6 AFAIK is a max bytes count of UTF8 characters

Reply:
fixed